server/auth.go

```diff
@@ -45,3 +45,84 @@ func AuthorizedToken(auth oauth2.Authenticator, logger chronograf.Logger, next h
 		return
 	})
 }
+
+// AuthorizedUser extracts the user name and provider from context. If the user and provider can be found on the
+// context, we look up the user by their name and provider. If the user is found, we verify that the user has at
+// at least the role supplied.
+func AuthorizedUser(store chronograf.UsersStore, useAuth bool, role string, logger chronograf.Logger, next http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !useAuth {
+			next(w, r)
+			return
+		}
+
+		log := logger.
+			WithField("component", "role_auth").
+			WithField("remote_addr", r.RemoteAddr).
+			WithField("method", r.Method).
+			WithField("url", r.URL)
+
+		ctx := r.Context()
+
+		username, err := getUsername(ctx)
+		if err != nil {
+			log.Error("Failed to retrieve username from context")
+			Error(w, http.StatusUnauthorized, "User is not authorized", logger)
+			return
+		}
+		provider, err := getProvider(ctx)
+		if err != nil {
+			log.Error("Failed to retrieve provider from context")
+			Error(w, http.StatusUnauthorized, "User is not authorized", logger)
+			return
+		}
+
+		u, err := store.Get(ctx, chronograf.UserQuery{Name: &username, Provider: &provider})
+		if err != nil {
+			log.Error("Error to retrieving user")
+			Error(w, http.StatusUnauthorized, "User is not authorized", logger)
+			return
+		}
+
+		if hasPrivilege(u, role) {
```

Comment:
we don't use the word `privilege` anywhere else in our authorization scheme, so i'd prefer to call this helper func something consistent (and maybe more precise), such as `hasAuthorizedRole`

Reply:
Fixed